tests/utils/index.ts

```diff
@@ -86,6 +137,33 @@ export const getSnapshotPath = (
       }
     }
   }
-
   return specificVersionSnapshotName;
 };
+
+export const getVersionsForTest = (
+  dirname: string,
+  linterName: string,
+  prefix: string,
+  checkType: "check" | "fmt"
+) => {
+  // TODO(Tyler): Add ARGS.linterVersion Query case for full matrix coverage
+  if (ARGS.linterVersion !== "Snapshots") {
+    return [undefined];
+  }
+
+  const versionsList = fs
+    .readdirSync(path.resolve(dirname, TEST_DATA))
+    .map((file) => {
+      const fileMatch = file.match(getSnapshotRegex(linterName, prefix, checkType));
+      if (fileMatch) {
+        return fileMatch.groups?.version;
+      }
+    })
+    .filter(Boolean);
+
+  if (versionsList.length === 0) {
+    return [undefined];
+  }
+
+  return new Set(versionsList);
```

Comment:
can we just return `versionsList` so that we always test in the same order?

Reply:
There's some de-duping that is happening here in the case of multiple prefixes. I'll add some logic for this for deterministic ordering, and also a comment explaining why

Reply:
Nevermind, I restructured in order to pass in a prefix. I'll just make this a list